src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +329,72 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 }
 
 
+/*
+ * undistribute_table gets a distributed table name and 
+ * udistributes the table.
+ */
+Datum
+undistribute_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	Oid relationId = PG_GETARG_OID(0);
+	const char *relationName = get_rel_name(relationId);
+
+	EnsureTableOwner(relationId);
+
+	Relation relation = relation_open(relationId, ExclusiveLock);
+
+	StringInfo query = makeStringInfo();
+
+	SPI_connect();
+
+	List * tableDDLEvents = GetTableDDLEvents(relationId, true);
+	char * createTableQuery = linitial(tableDDLEvents);
+	list_delete_first(tableDDLEvents);
+
+	Node *parseTree = ParseTreeNode(createTableQuery);
+
+	((CreateStmt *)parseTree)->relation->relname = "tmp1";
```

Comment:
I think we should use table name with a randomized suffix. Otherwise this will fail if someone has a "tmp1" table already. 

Something like `{oldRelationName}_tmp_{thirtyRandomAlphaNumerics}`.

Reply:
You can take a look at how we solve too long table names after a shardId is appended in `AppendShardIdToName()` https://github.com/citusdata/citus/blob/c61e84c14b674af2cabadd6ad7ae19c282f2e5bd/src/backend/distributed/relay/relay_event_utility.c#L672-L678

It should give you an idea on how to safely create table names